cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/ToDoListWidget.kt

```diff
@@ -0,0 +1,58 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.storage.ContentRepository
+import com.cognifide.cogboard.widget.BaseWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+
+class ToDoListWidget(vertx: Vertx, config: JsonObject) : BaseWidget(vertx, config) {
+
+    init {
+        createDynamicChangeSubscriber()?.handler {
+            val state = changeSelectedItems(it.body())
+            send(state)
+        }
+    }
+
+    override fun updateState() {
+        val selectedItems = selectedItems()
+        when {
+            selectedItems != null -> {
+                send(contentJson(selectedItems))
+            }
+            else -> {
+                send(contentJson())
+            }
+        }
+    }
+
+    private fun changeSelectedItems(state: JsonObject): JsonObject {
+        val selectedItems = selectedItems()
+        val selectedItem = state.getValue(SELECTED_ITEM)
+        when {
+            selectedItems?.contains(selectedItem)!! -> {
+                selectedItems.remove(selectedItem)
+            }
+            else -> {
+                selectedItems.add(selectedItem)
+            }
+        }
+        return contentJson(selectedItems)
```

Comment:
This method is a little bit fishy:
1. make sure that `selectedItems` does not return null use elvis `?: JsonArray()`
2. change when to if/else if there are only 2 possible outcomes